ee/tabby-webserver/src/schema/email_service_credential.rs

```diff
@@ -12,7 +12,11 @@ pub struct EmailServiceCredential {
 #[async_trait]
 pub trait EmailServiceCredentialService: Send + Sync {
     async fn get_email_service_credential(&self) -> Result<Option<EmailServiceCredential>>;
-    async fn update_email_service_credential(&self, creds: EmailServiceCredentialDAO)
-        -> Result<()>;
+    async fn update_email_service_credential(
+        &self,
+        smtp_username: String,
+        smtp_password: String,
```

Comment:
```suggestion
        smtp_password: Optional<String>,
```

Since we do not pass password to client side, we need to support updates without password, so users could change SMTP server / protocol without having to retype the password